File: python/bad_examples/type_violation/type_violation.py
"""Deliberately type-unsafe code: wrong argument types and untyped parameters.

The types gate must reject both patterns. See bad_examples/README.md for the
manifest of expected signals.
"""


def add_units(left: int, right: int) -> int:
    """Add two unit counts."""
    return left + right


total = add_units("3", 4)


def combine(raw_left, raw_right):
    """Parameters carry no annotations at all."""
    return raw_left + raw_right

File: typescript/bad_examples/type_violation/type_violation.ts
/**
 * Deliberately type-unsafe code: a wrong-type initializer and unannotated
 * parameters.
 *
 * The types gate must reject both patterns; assert.sh scopes `tsc -p` at
 * this directory via its own tsconfig.json and expects `error TS2322` among
 * the diagnostics. See bad_examples/README.md for the manifest.
 */

const unitsTotal: number = "3";

function combine(rawLeft, rawRight): number {
  const left = rawLeft;
  const right = rawRight;
  return left + right;
}

export const combineUnits = combine;
